### PR TITLE
Domzippilli/add flatten flag

### PR DESCRIPTION
**Pull request by domZippilli**
Changes:

* Adds flag parsing using `flag`, including `-flatten` flag
* Adds flattening logic, wherein the module names are prepended to the instance name instead of making the instances children of the modules in the tree
* Update tests to include both flat and hierarchical cases

Also, the test TF was changed to use GCP as it was more convenient than trying to get AWS account + creds to run a plan; feel free to revert if you have an AWS account handy to run the test & tweak.

### CODE REVIEW

File: converter/util.go [hunk 1]
```diff
@@ -21,10 +25,25 @@ func insert(out output, path []string, key string, value interface{}) {
...
+			if len(diff.Path) > 1 {
+				// slice [1:] to omit "Root" string
+				flatName.WriteString(strings.Join(diff.Path[1:], "."))
+				flatName.WriteString(".")
+			}
+
+			flatName.WriteString(k)
+			instanceName = []string{flatName.String()}
+		} else {
+			insert(out, diff.Path, "destroy", diff.Destroy)
```
Comment (domZippilli) [inline converter/util.go:49]: @morgante in the flattened form, each of the instances has the `"destroy": false"` pair, so I omitted it from the root level as it seemed redundant. If that breaks compatibility with consuming tools, I'll add.

**Comment by domZippilli** [comment]
Only thing I'm left wondering is if it's useful to test more complex Terraform code. The tests use a pretty sparse input and don't use a ton of TF features. I suspect, though, that none of those features actually produce anything novel in the plan "backend" output. It probably all gets reconciled in the plan compiler -- dependencies, variables, that sort of thing. Thoughts?

File: converter/util.go (hunk 1 above)
Comment (morgante) [reply converter/util.go:49]: @domZippilli What do you mean by that? Having info on whether an individual resource is being destroyed or not would be very critical for certain plan analysis.

**Comment by morgante** [comment]
@domZippilli Could you file a separate PR with the AWS->TF conversion for testing?

File: converter/util.go (hunk 1 above)
Comment (domZippilli) [reply converter/util.go:49]: Sure. That's not at all what I said. It's still transcribed for each instance. This is just about the root of the flat tree. Shouldn't be hard to add back.

**Comment by domZippilli** [comment]
> @domZippilli Could you file a separate PR with the AWS->TF conversion for testing?

Do you mean AWS->GCP? Like, you want the test code to still use AWS?

File: converter/util.go (hunk 1 above)
Comment (domZippilli) [reply converter/util.go:49]: I added it to the flattened tree. It just takes a little logic so you don't end up with objects for each module that just has a destroy pair and nothing else.

**Comment by morgante** [comment]
I meant AWS to GCP. Changing the test system should ideally be independent
of any code changes.

**Comment by domZippilli** [comment]
K

**Comment by domZippilli** [comment]
ack

**Pull request closed without merging**